plc4j/protocols/s7/src/main/java/org/apache/plc4x/java/s7/netty/Plc4XS7Protocol.java

```diff
@@ -54,6 +54,17 @@ Licensed to the Apache Software Foundation (ASF) under one
 import static org.apache.plc4x.java.s7.netty.util.S7TypeDecoder.decodeData;
 import static org.apache.plc4x.java.s7.netty.util.S7TypeEncoder.encodeData;
 
+/**
+ * This layer transforms between {@link PlcRequestContainer}s {@link S7Message}s.
+ * And stores all "in-flight" requests in an internal structure ({@link Plc4XS7Protocol#requests}).
+ *
+ * TODO 01.08.18, jf: is a Cleanup of the requests map necessary? Can it occur that for a request no response is generated that leads to success or failure?
```

Comment:
Yes we should have some housekeeping mechanism in place that handles timeouts and connection problems. The driver implementation we are using in the Enthernet/IP driver (https://github.com/digitalpetri/ethernet-ip) has been a great inspiration on how to do this. We just haven't done it yet. Perhaps it would be a good idea to create a Jira for this?